diff -bur moodle-unmod/user/edit.php moodle/user/edit.php
--- moodle-unmod/user/edit.php	2008-07-11 08:15:32.000000000 +0100
+++ moodle/user/edit.php	2008-09-15 13:44:54.000000000 +0100
@@ -183,9 +183,12 @@
     $strparticipants  = get_string('participants');
     $userfullname     = fullname($user, true);
     if ($course->id != SITEID) {
+        $parts='';
+        if (isteacher($course->id))
+           $parts='-> <a href=\"index.php?id=$course->id\">$strparticipants</a>';
         print_header("$course->shortname: $streditmyprofile", "$course->fullname: $streditmyprofile",
                      "<a href=\"$CFG->wwwroot/course/view.php?id=$course->id\">$course->shortname</a>
-                      -> <a href=\"index.php?id=$course->id\">$strparticipants</a>
+                      $parts
                       -> <a href=\"view.php?id=$user->id&amp;course=$course->id\">$userfullname</a>
                       -> $streditmyprofile", "");
     } else {
Only in moodle/user/: edit.php.orig
diff -bur moodle-unmod/user/view.php moodle/user/view.php
--- moodle-unmod/user/view.php	2008-07-06 01:15:24.000000000 +0100
+++ moodle/user/view.php	2008-12-08 16:20:13.000000000 +0000
@@ -4,6 +4,7 @@
     
     require_once("../config.php");
     require_once($CFG->dirroot.'/user/profile/lib.php');
+    require_once("../blocks/pageone/pageonelib.php");
     $id      = optional_param('id',     0,      PARAM_INT);   // user id
     $course  = optional_param('course', SITEID, PARAM_INT);   // course id (defaults to Site)
     $enable  = optional_param('enable', '');                  // enable email
@@ -439,18 +440,33 @@
             }
             echo "<form onclick=\"this.target='message'\" action=\"../message/index.php\" method=\"get\">";
             echo "<div>";
-            echo "<input type=\"submit\" value=\"$messagebuttonname\" onclick=\"return openpopup('/message/index.php', 'message', 'menubar=0,location=0,scrollbars,status,resizable,width=400,height=500', 0);\" />";
+            echo "<input type=\"submit\" value=\"$messagebuttonname\" onclick=\"return openpopup('/message/index.php', 'message', 'menubar=0,location=0,scrollbars,status,resizable,width=400,height=600', 0);\" />";
             echo "</div>";
             echo "</form>";
         } else {
             echo "<form onclick=\"this.target='message$user->id'\" action=\"../message/discussion.php\" method=\"get\">";
             echo "<div>";
             echo "<input type=\"hidden\" name=\"id\" value=\"$user->id\" />";
-            echo "<input type=\"submit\" value=\"".get_string("sendmessage", "message")."\" onclick=\"return openpopup('/message/discussion.php?id=$user->id', 'message_$user->id', 'menubar=0,location=0,scrollbars,status,resizable,width=400,height=500', 0);\" />";
+            echo "<input type=\"submit\" value=\"".get_string("sendmessage", "message")."\" onclick=\"return openpopup('/message/discussion.php?id=$user->id', 'message_$user->id', 'menubar=0,location=0,scrollbars,status,resizable,width=400,height=600', 0);\" />";
             echo "</div>";
             echo "</form>";
         }
     }
+
+    // MoodleMobile patch
+    if ($user->id!=$USER->id)
+    {
+        $courseid=pageone_find_course($user->id, $USER->id);
+        if ($courseid!=null)
+        {
+            echo "<form action=\"../blocks/pageone/email.php\" method=\"get\">".
+                 "<input type=\"hidden\" name=\"to\" value=\"".$user->id."\" />".
+                 "<input type=\"hidden\" name=\"id\" value=\"".$courseid."\" />".
+                 "<input type=\"submit\" value=\"".get_string("contact_with_mm", "block_pageone")."\" />".
+                 "</form>";
+        }
+    }
+
     // Authorize.net: User Payments
     if ($course->enrol == 'authorize' || (empty($course->enrol) && $CFG->enrol == 'authorize')) {
         echo "<form action=\"../enrol/authorize/index.php\" method=\"get\">";
